src/Database/Schema/ManagedTable.php

```diff
@@ -0,0 +1,116 @@
+<?php
+	/*
+	* Copyright (c) 2017 Morten Nilsen (morten@runsafe.no)
+	* https://github.com/Kruithne/KrameWork7
+	*
+	* Permission is hereby granted, free of charge, to any person obtaining a copy
+	* of this software and associated documentation files (the "Software"), to deal
+	* in the Software without restriction, including without limitation the rights
+	* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+	* copies of the Software, and to permit persons to whom the Software is
+	* furnished to do so, subject to the following conditions:
+	*
+	* The above copyright notice and this permission notice shall be included in all
+	* copies or substantial portions of the Software.
+	*
+	* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+	* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+	* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+	* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+	* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+	* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+	* SOFTWARE.
+	*/
+
+	namespace KrameWork\Database\Schema;
+
+	require_once(__DIR__ . 'IManagedTable.php');
+	require_once(__DIR__ . 'IMetaTable.php');
+	require_once(__DIR__ . '../Driver/Generic.php');
+
+	use \KrameWork\Database\Driver\Generic;
+
+	/**
+	 * Class to handle a managed table hosted by a Microsoft SQL Server
+	 * @author docpify <morten@runsafe.no>
+	 * @package KrameWork\Database
+	 */
+	abstract class ManagedTable implements IManagedTable
+	{
+		public function __construct(Generic $db, IMetaTable $metaTable) {
+			$this->db = $db;
+			$this->meta = $metaTable;
+		}
+
+		/**
+		 * @api update
+		 * Updates the table in the database
+		 * @return string Status
+		 */
+		public function update() {
+			$current = $this->meta->getVersion($this->GetName());
+			if($current < 1)
+				return $this->create();
+
+			$log = $this->versionLog();
+			if($current > count($log))
+				return 'Table is ahead of code: '.$current;
+
+			if($current == count($log))
```

Comment:
Instead of calling count() here and in line 56, it should be saved to a local variable to save a function call, tho opcache can optimize it away